src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetRelevantDeparmentProfile.cs

```diff
@@ -0,0 +1,322 @@
+using Fusion.Integration.Profile;
+using Fusion.Integration.Roles;
+using Fusion.Resources.Domain.Models;
+using Fusion.Services.LineOrg.ApiModels;
+using MediatR;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using Microsoft.MarkedNet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+
+namespace Fusion.Resources.Domain.Queries
+{
+    /// <summary>
+    /// Fetch the profile for a resource owner. Will compile a list of departments the person has responsibilities in and which is relevant.
+    /// </summary>
+    public class GetRelevantDeparmentProfile : IRequest<IEnumerable<QueryRelevantDepartmentProfile>>
+    {
+        public GetRelevantDeparmentProfile(string profileId, AspNetCore.OData.ODataQueryParams query)
+        {
+            ProfileId = profileId;
+            Query = query;
+        }
+
+        /// <summary>
+        /// Mail or azure unique id
+        /// </summary>
+        public PersonId ProfileId { get; set; }
+        public ODataQueryParams Query { get; }
+
+
+
+        public class Handler : IRequestHandler<GetRelevantDeparmentProfile, IEnumerable<QueryRelevantDepartmentProfile>>
+        {
+            private readonly TimeSpan defaultAbsoluteCacheExpirationHours = TimeSpan.FromHours(1);
+            private readonly ILogger<Handler> logger;
+            private readonly IFusionProfileResolver profileResolver;
+            private readonly IFusionRolesClient rolesClient;
+            private readonly IMediator mediator;
+            private readonly ILineOrgResolver lineOrgResolver;
+            private readonly IMemoryCache memCache;
+            private bool sapIdFound = false;
+            List<QueryRelevantDepartmentProfile?> lstDepartments = new List<QueryRelevantDepartmentProfile?>();
+
+            public Handler(ILogger<Handler> logger, IFusionProfileResolver profileResolver, IFusionRolesClient rolesClient, ILineOrgResolver lineOrgResolver, IMediator mediator, IMemoryCache memCache)
+            {
+                this.logger = logger;
+                this.profileResolver = profileResolver;
+                this.rolesClient = rolesClient;
+                this.mediator = mediator;
+                this.lineOrgResolver = lineOrgResolver;
+                this.memCache = memCache;
+
+            }
+            private CancellationToken _cancellationToken;
+            public async Task<IEnumerable<QueryRelevantDepartmentProfile>> Handle(GetRelevantDeparmentProfile request, CancellationToken cancellationToken)
+            {
+                _cancellationToken = cancellationToken;
+                var user = await profileResolver.ResolvePersonBasicProfileAsync(request.ProfileId.OriginalIdentifier);
+
+                if (user is null) return null;
```

Comment:
When expecting a list, you return null here. 
Check with similar list-queries :-)